float32 posx
float32 posy
bool picking
bool releasing
---
bool success
---
float32 progress